### PR TITLE
Feature/smpp channel

### PR DESCRIPTION
**Pull request by ihor-palii**
https://communityconnectlabs.atlassian.net/jira/software/projects/CD/boards/3?selectedIssue=CD-37
https://communityconnectlabs.atlassian.net/jira/software/projects/CD/boards/3?selectedIssue=CD-38

### CODE REVIEW

**Review by musamusa — APPROVED** [review]

**Review by teehamaral — COMMENTED** [review]
After our call today, I think we will need a different approach for the UI, likely, it's going to be set up via the settings file and attached to the organization by the implementation team somehow.

File: temba/msgs/models.py [hunk 1]
```diff
@@ -1540,3 +1542,10 @@ class MessageExportAssetStore(BaseExportAssetStore):
     directory = "message_exports"
     permission = "msgs.msg_export"
     extensions = ("xlsx",)
+
+
+class MessageExternalIDMap(models.Model):
+    message = models.ForeignKey(Msg, on_delete=models.CASCADE, related_name="external_ids")
+    channel = models.ForeignKey(Channel, on_delete=models.CASCADE)
+    carrier_id = models.CharField(max_length=64, blank=True, null=True)
```
Comment (teehamaral) [inline temba/msgs/models.py:1550]: Quick question: could this field be an integer? Or does it need to be a string?

Comment (teehamaral) [reply temba/msgs/models.py:1550]: Also, do you think that we should have the created_at and updated_at fields?

Comment (ihor-palii) [reply temba/msgs/models.py:1550]: > @ihor-palii - Quick question: could this field be an integer? Or does it need to be a string?

In mGage docs, this id could contain characters so I that is why I made it CharField (per what I have tested I saw only numbers).

Comment (ihor-palii) [reply temba/msgs/models.py:1550]: > Also, do you think that we should have the created_at and updated_at fields?

That's, a good Idea I will add these two fields.

Comment (teehamaral) [reply temba/msgs/models.py:1550]: > > @ihor-palii - Quick question: could this field be an integer? Or does it need to be a string?
> 
> In mGage docs, this id could contain characters so I that is why I made it CharField (per what I have tested I saw only numbers).

Got it, so let's keep CharField :)

**Review by teehamaral — APPROVED** [review]

**Pull request merged**